src/execution_plan.c

```diff
@@ -642,6 +650,15 @@ static void ExecutionPlan_Distribute(ExecutionPlan* ep){
     Gears_BufferWriter bw;
     Gears_BufferWriterInit(&bw, buff);
     size_t len;
+        if(EPIsFlagOn(ep->fep, EFRegistered)){
+        // Registered execution plan - serialize id and return.
+        RedisGears_BWWriteLong(&bw, 1);
+        FlatExecutionPlan_SerializeID(ep->fep, &bw);
+        return;
+    } else {
+        RedisGears_BWWriteLong(&bw, 0); // Non Registered execution plan.
+    }
+    
     int res = FlatExecutionPlan_Serialize(&bw, ep->fep, NULL);
```

Comment:
Notice that you serialize the FEP anyway

Reply:
Do I need to serialize? 
I thought that we just serializing ID

Reply:
Yes but this line happened anyway in you code:
FlatExecutionPlan_Serialize(...

Reply:
Ohh sorry, missed the return ..

Reply:
But this is an issue, if you return you do not serialize the reader data. This is why the tests fails ..